### PR TITLE
Some enhancements

### PR DESCRIPTION
**Pull request by bgaifullin**
- added messages for test asserts
- modified condition to select expired jobs instead of exact match
- optimisation: did re-create timer on request for snapshot

### CODE REVIEW

**Comment by bgaifullin** [comment]
@robfig could you please review this PR ?

File: cron_test.go [hunk 1]
```diff
@@ -221,7 +220,7 @@ func TestLocalTimezone(t *testing.T) {
 	wg := &sync.WaitGroup{}
 	wg.Add(2)
 
-	now := time.Now().Local()
+	now := time.Now()
```
Comment (robfig) [inline cron_test.go]: what is the purpose of this change?

Comment (bgaifullin) [inline cron_test.go]: since time.Now() is local time already, so it is not need to call .Local()

**Comment by robfig** [comment]
Ok, thanks!
